lib/instrumentation/modules/memcached.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+var semver = require('semver')
+
+var shimmer = require('../shimmer')
+
+module.exports = function (memcached, agent, { version, enabled }) {
+  if (!enabled) {
+    return memcached
+  }
+  if (!semver.satisfies(version, '>=2.2.0')) {
+    agent.logger.debug('Memcached version %s not supported - aborting...', version)
+    return memcached
+  }
+  agent.logger.debug('shimming memcached.prototype.command')
+  shimmer.wrap(memcached.prototype, 'command', wrapCommand)
+  return memcached
+
+  // Wrap the generic command that is used to build touch, get, gets etc
+  function wrapCommand (original) {
+    return function wrappedCommand () {
+      if (typeof arguments[0] === 'function') {
+        var query = arguments[0]()
+        // If the callback is not a function the user doesn't care about result
+        if (query && typeof query.callback === 'function') {
+          var span = agent.startSpan(`memcached.${query.type}`, `db.memcached.${query.type}`)
+          if (span) {
+            span.setDbContext({ statement: `${query.type} ${query.key}`, type: 'memcached' })
+            agent.logger.debug('intercepted call to memcached.prototype.command %o', { id: span.id, type: query.type })
```

Comment:
I know this probably wasn't that clear, but it was actually on purpose that this debug log statement was logged even if no span was created. This better allows us to debug the flow of the problem even if spans are not generated. That's why the `span.id` was put into a variable that would then either be `undefined` or contain the spna id.

But I do like that we don't have to wrap the callback if no span is created. That's a clear improvement.

How about we just move this line outside of the `if (span)` block (just before) and change it accordingly:

```diff
-            agent.logger.debug('intercepted call to memcached.prototype.command %o', { id: span.id, type: query.type })
+            agent.logger.debug('intercepted call to memcached.prototype.command %o', { id: span && span.id, type: query.type })
```